### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by andriy-shymkiv**

### CODE REVIEW

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+    const length = this.length;
+
+    for (let i = 0; i < this.length; i++) {
```
Comment (iryna-lelyk) [inline src/arrayMethodJoin.js:11]: Why don't you use here your variable 'length'?

**Review by iryna-lelyk — APPROVED** [review]
Looks good to me.

File: src/arrayMethodJoin.js (hunk 1 above)
Comment (Andriy-Shymkiv) [reply src/arrayMethodJoin.js:11]: good question ))

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+    const length = this.length;
```
Comment (illiarerun) [inline src/arrayMethodJoin.js:9]: don't create redundant variables
```suggestion
```

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
```
Comment (illiarerun) [inline src/arrayMethodJoin.js:8]: don't use names like that, try to come up with more meaningful name

File: src/arrayMethodJoin.js [hunk 4]
```diff
@@ -4,8 +4,24 @@
...
+    const length = this.length;
+
+    for (let i = 0; i < this.length; i++) {
+      if (typeof this[i] === 'undefined'
+      || this[i] === null) {
+        this[i] = '';
+      }
```
Comment (illiarerun) [inline src/arrayMethodJoin.js:15]: Why don't you just add `this[i]` to string if it's not `null` or undefined, you're repeating yourself in next `if` block - try to invert your if statement to do that. Also, you don't need `typeof` here.

File: src/arrayMethodJoin.js [hunk 5]
```diff
@@ -4,8 +4,24 @@
...
+
+      if (i !== length - 1) {
+        string += this[i] + String(separator);
+      } else {
+        string += this[i];
+      }
```
Comment (illiarerun) [inline src/arrayMethodJoin.js:21]: and then you'd only add separator here like that
```suggestion
      if (i !== length - 1) {
        string += String(separator);
      } 
```

**Review by illiarerun — CHANGES_REQUESTED** [review]

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**